### PR TITLE
Update version to 0.1b1

### PR DESCRIPTION
**Pull request by cretz**
## What was changed

Update version and add minor note to README
## Why?

Beta release

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "temporalio"
-version = "0.1a2"
+version = "0.1b1"
```
Comment (mmcshane) [inline pyproject.toml:3]: Kind of curious why we don't use semver - is it not widely used in the Python world? FWIW, I think most of the promises of semver are BS but at least it has a formalized syntax

**Review by mmcshane — APPROVED** [review]
delivers the intent, which is "beta 1"

File: pyproject.toml (hunk 1 above)
Comment (cretz) [reply pyproject.toml:3]: We will use semver upon GA. This is Python's knowingly-non-semver form of expressing alpha/beta. See https://peps.python.org/pep-0440/ (specifically https://peps.python.org/pep-0440/#semantic-versioning and discussion therein about prerelease builds).

Comment (mmcshane) [reply pyproject.toml:3]: Cooll, thanks for the link.

**Pull request merged**